### PR TITLE
feat: Add `charge` method to the run client for "pay per event"

### PR DESCRIPTION
**Pull request by Jkuzz**
Resolves https://github.com/apify/apify-core/issues/18592 by adding the PPE charge endpoint to JS client.

The idempotency key creation was taken from an actor by the store team.
Issue to add docs URL https://github.com/apify/apify-client-js/issues/614

### CODE REVIEW

File: src/resource_clients/run.ts [hunk 1]
```diff
@@ -138,6 +141,35 @@ export class RunClient extends ResourceClient {
         return cast(parseDateFields(pluckData(response.data)));
     }
 
+    /**
+     * TODO: docs url
+     * https://github.com/apify/apify-client-js/issues/614
```
Comment (B4nan) [inline src/resource_clients/run.ts:146]: can we add the URL here right ahead? even if it doesn't exist yet, feels better than publishing with a TODO

**Review by B4nan — APPROVED** [review]
you can ignore the failing docs build, its not detecting the links inside parens correctly

File: src/resource_clients/run.ts (hunk 1 above)
Comment (Jkuzz) [reply src/resource_clients/run.ts:146]: I don't have the URL. I'll ask Mat'o what's the status on the endpoint docs 👍

Comment (B4nan) [reply src/resource_clients/run.ts:146]: I am rather sure we can guess it before it exists, its just another endpoint, the URLs for them are not random.

File: src/resource_clients/run.ts [hunk 2]
```diff
@@ -138,6 +141,34 @@ export class RunClient extends ResourceClient {
         return cast(parseDateFields(pluckData(response.data)));
     }
 
+    /**
+     * https://docs.apify.com/api/v2#/reference/actor-runs/charge-run/charge-run
+     */
+    async charge(options: RunChargeOptions): Promise<ApifyResponse<Record<string, never>>> {
```
Comment (barjin) [inline src/resource_clients/run.ts:147]: Out of curiosity - what's the expected return type of this method? I see that most of the `RunClient` methods return `ActorRun` object (which reflects the changes made by the call). 

Is this return type (`ApifyResponse<Record<string, never>>`) a temporary thing before we figure out the actual API response shape, or is this the final thing?

Comment (Jkuzz) [reply src/resource_clients/run.ts:147]: We discussed this in person. The endpoint uses a memorized version of the run from the database for performance reasons, as the run's pricing model cannot change during the run. Therefore we cannot return the modified run object and the only important output is the response status code.

File: src/resource_clients/run.ts [hunk 3]
```diff
@@ -138,6 +141,34 @@ export class RunClient extends ResourceClient {
...
+        ow(options, ow.object.exactShape({
+            eventName: ow.string,
+            count: ow.optional.number,
+            idempotencyKey: ow.optional.string,
+        }));
+
+        const count = options.count ?? 1;
+        const idempotencyKey = options.idempotencyKey ?? `${this.id}-${options.eventName}-${Date.now()}`;
```
Comment (fnesveda) [inline src/resource_clients/run.ts:155]: Same note here as in the Python client, the idempotency key has to be a unique string. This way, if someone charged for two occurences of the same event in the same millisecond, only one of the charges would go through.

https://github.com/apify/apify-client-python/pull/304#pullrequestreview-2480812542

**Review by fnesveda — CHANGES_REQUESTED** [review]
One problem copied over from the Python client review

**Review by fnesveda — APPROVED** [review]

**Pull request merged**